### ISSUE TITLE
PTBImporter appends spaces to all tokens

### ISSUE THREAD

**Issue by adrianeboyd** [korpling/pepperModules-PTBModules#1]
The PTBImporter appends one or two spaces to all tokens in the corpus.

I think part of the problem might be in this section of PTBImporterMapper, which sets the token length with the appended space rather than without:

```
// put strTok + " " into STextualDS
intTokStartChar = stbText.toString().length();
stbText.append(strTok);
stbText.append(" ");
intTokEndChar = stbText.toString().length();

// create SToken node covering the strTok position
SToken tokCurrentToken = getSDocument().getSDocumentGraph().createSToken(txtText, intTokStartChar, intTokEndChar);
```

About a third of the tokens in the converted corpus ended up with two appended spaces and I'm not sure where the second space might be getting inserted.

**Comment by amir-zeldes** [comment]
The PTB importer appends spaces to tokens in Salt to create a ‘readable’ primary STextualDS. This is done because PTB bracketing is non-whitespace preserving (there is no real way of expressing things other than tokens, tags and non-terminal nodes/edges), but some output formats of SaltNPepper are, such as PAULA XML. 

Because intelligent space insertion (i.e. guessing that before words and after punctuation we want a space, but not between a word and punctuation) is bound to fail sometimes, it was decided to always include a space between tokens so that a readable text is formed. In non-whitespace preserving output formats, this information is lost again and is irrelevant. 

I’m not sure where your second space is coming from – are you sure it’s being generated by the importer and not a subsequent exporter? Is there some way that spaces could be included at the end of a line and appended to the tokens as part of a token? Perhaps some Windows/Unix/Mac end of line symbol issue?

**Comment by adrianeboyd** [comment]
Hi Amir,

I understand that you need to insert spaces in the text and I think
it's fine to just add spaces between all tokens, but I don't think they
need to be in the tokens themselves. In ANNIS, if you want to search for
the word "the" you have to search for "the " instead, which does not
seem quite right.  In node.tab all the token lines have one or two
spaces at the end after my very plain SnP PTBImporter/relANNIS exporter
conversion.

I think that the token creation here:

SToken tokCurrentToken =
getSDocument().getSDocumentGraph().createSToken(txtText,
intTokStartChar, intTokEndChar);

should be:

SToken tokCurrentToken =
getSDocument().getSDocumentGraph().createSToken(txtText,
intTokStartChar, intTokEndChar - 1);

I haven't tracked down where the second spaces were coming from.  I can
send you a sample corpus, though, if you'd like.  My first guess would
be that one of the .trim()s isn't working as expected, maybe because of
line breaks characters like you mentioned.

Thanks,
Adriane

**Comment by amir-zeldes** [comment]
Oh, OK, I understand. That is indeed strange behavior, the space should only be in the textual DS. I’ll look into it.

**Comment by amir-zeldes** [comment]
Hi Adriane,

I’ve taken over the fix you suggested (sorry for the delay). I still don’t know about the second spaces, so if they’re still a problem, please send me an example to debug. 

Amir

**Comment by amir-zeldes** [comment]
Fixed in 7be5d6f7eaf84c22a0d019b3b6a754a0dfe67bfd
